api/pkg/transformer/pipeline/compiler.go

```diff
@@ -198,6 +205,77 @@ func (c *Compiler) parseTablesSpec(tableSpecs []*spec.Table, compiledJsonPaths *
 	return NewCreateTableOp(tableSpecs), nil
 }
 
+func (c *Compiler) parseJsonOutputSpec(jsonSpec *spec.JsonOutput, compiledJsonPaths *jsonpath.Storage, compiledExpressions *expression.Storage) (Op, error) {
+	template := jsonSpec.JsonTemplate
+	if template == nil {
+		return nil, errors.New("jsontemplate must be specified")
+	}
+	if template.BaseJson != nil {
+		compiledJsonPath, err := jsonpath.Compile(template.BaseJson.JsonPath)
+		if err != nil {
+			return nil, err
+		}
+		compiledJsonPaths.Set(template.BaseJson.JsonPath, compiledJsonPath)
+	}
+
+	if err := c.parseJsonFields(template.Fields, compiledJsonPaths, compiledExpressions); err != nil {
+		return nil, err
+	}
+
+	jsonOutputOp := NewJsonOutputOp(jsonSpec)
+
+	return jsonOutputOp, nil
+}
+
+func (c *Compiler) parseJsonFields(fields map[string]*spec.Field, compiledJsonPaths *jsonpath.Storage, compiledExpressions *expression.Storage) error {
+	for _, field := range fields {
+		switch val := field.Value.(type) {
+		case *spec.Field_FromJson:
+			if len(field.Fields) > 0 {
+				return errors.New("can't specify nested json, if field has value to set")
+			}
+			compiledJsonPath, err := jsonpath.Compile(val.FromJson.JsonPath)
+			if err != nil {
+				return err
+			}
+			compiledJsonPaths.Set(val.FromJson.JsonPath, compiledJsonPath)
+		case *spec.Field_FromTable:
+			if len(field.Fields) > 0 {
+				return errors.New("can't specify nested json, if field has value to set")
+			}
+
+			// Check whether format specified is valid or not
+			if val.FromTable.Format == spec.FromTable_INVALID {
+				return errors.New("table format specified is invalid")
+			}
+
+			// Check whether table name already registered or not
+			tableName := val.FromTable.TableName
```

Comment:
I think we can just simply check whether the variable is declared and doesn't need to treat it as expression